shared/middleware/files_api.rb

```diff
@@ -24,6 +26,10 @@ def get_bucket_impl(endpoint)
 
   def allowed_file_type?(endpoint, extension)
     case endpoint
+    when 'animations'
+      # Only allow specific image types to be uploaded by users.
+      # Only png for now, will expand support in the future
+      %w(.png).include? extension.downcase
```

Comment:
This switch statement "smells" like it belongs in the bucket implementation. Extra credit if you agree and want to move it there. Supporting evidence: http://c2.com/cgi/wiki?SwitchStatementsSmell "Typically, similar switch statements are scattered throughout a program"